### ISSUE TITLE
PointLocator robustness

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#4528]
http://sourceforge.net/p/libmesh/mailman/libmesh-devel/thread/CAFfxPjqYJNY6-1kTVqTsyMaOr1Jpu9QnMEWrp7swVdC0z8Oi2w%40mail.gmail.com/#msg33221196

These issues have also forced us to be more careful about using PointLocator in parallel regions where different processors can get different points. See #4455 for places in the framework where this has been an issue in the past.

**Comment by permcody** [comment]
Closing in favor of https://github.com/libMesh/libmesh/issues/432
